edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet.Docker/SystemResourcesMetrics.cs

```diff
@@ -215,16 +221,23 @@ Option<double> GetCpuUsage(DockerStats module)
         Option<double> GetCpuLinux(DockerStats module)
         {
             Option<double> currentTotal = module.CpuStats.AndThen(s => s.CpuUsage).AndThen(s => s.TotalUsage);
-            Option<double> previousTotal = module.PreviousCpuStats.AndThen(s => s.CpuUsage).AndThen(s => s.TotalUsage);
+            Option<double> previousTotal = module.Name.AndThen(previousModuleCpu.GetOption);
             Option<double> moduleDelta = currentTotal.AndThen(curr => previousTotal.Map(prev => curr - prev));
 
             Option<double> currentSystem = module.CpuStats.AndThen(s => s.SystemCpuUsage);
-            Option<double> previousSystem = module.PreviousCpuStats.AndThen(s => s.SystemCpuUsage);
+            Option<double> previousSystem = module.Name.AndThen(previousSystemCpu.GetOption);
             Option<double> systemDelta = currentSystem.AndThen(curr => previousSystem.Map(prev => curr - prev));
 
+            // set previous to new current
+            module.Name.ForEach(name =>
+            {
+                currentTotal.ForEach(curr => previousModuleCpu[name] = curr);
+                currentSystem.ForEach(curr => previousSystemCpu[name] = curr);
+            });
+
             return moduleDelta.AndThen(moduleDif => systemDelta.AndThen(systemDif =>
             {
-                if (moduleDif > 0 && systemDif > 0)
+                if (moduleDif >= 0 && systemDif > 0)
```

Comment:
What's the reason for adding the "="? I saw the docker golang version calculated the percpuusage. Should we consider it?

Reply:
If it only counts a value when cpu > 0, it skews the average b/c it skips all the 0 values. So  if the readings were 0, 1, 0, 0, 1, 1 the average would be 1 instead of .5.